packages/common/src/ide/types/ide.types.ts

```diff
@@ -94,6 +94,12 @@ export interface IDE {
     listener: (event: TextDocumentChangeEvent) => void,
   ): Disposable;
 
+  /**
+   * Find occurrences of query string in the active document.
+   * @param query The string query to search for
+   */
+  findInDocument(query: string): Promise<void>;
+
```

Comment:
Tempted to have the abstraction accept a text editor as input 🤔

Reply:
Oh to open find widget in arbitrary splits?

Reply:
Yeah, just feels like possibly a better abstraction. I don't feel strongly tho

Reply:
I agree.

Reply:
I guess prob will be optional anyway so could possibly leave it off for now until we need it

Reply:
The question is if you target something in a split do you want to search that document or the active?

Reply:
If we pass the editor as an argument we can change this behavior in the future without changing the api. If we don't pass the editor we have no control over this.

Reply:
Yeah I'm happy for us to add it. Should be trivial, no?

Fwiw I'd lean towards searching in document containing cursor, but I could see either way

Reply:
Fixed. For now we default to the active editor, but in the future we could change this behavior easily.